parts/kuberneteswindowssetup.ps1

```diff
@@ -0,0 +1,299 @@
+    $MasterIP = "10.240.255.5",
+
+    [parameter()]
+    [ValidateNotNullOrEmpty()]
+    $KubeDnsServiceIp = "10.0.0.10",
+
+    [parameter()]
+    [ValidateNotNullOrEmpty()]
+    $CACertificate = "<<<caCertificate>>>",
+
+    [parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    $MasterFQDNPrefix,
+
+    [parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    $Location,
+
+    [parameter()]
+    [ValidateNotNullOrEmpty()]
+    $AgentCertificate = "<<<clientCertificate>>>",
+
+    [parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    $AgentKey,
+
+    [parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    $AzureHostname
+)
+
+$global:DockerServiceName = "Docker"
+$global:RRASServiceName = "RemoteAccess"
+$global:KubeDir = "c:\k"
+$global:KubeBinariesSASURL = "https://acsengine.blob.core.windows.net/v1-5-1/k.zip?st=2017-01-30T15%3A47%3A00Z&se=2020-01-31T15%3A47%3A00Z&sp=rl&sv=2015-12-11&sr=b&sig=C9ho0Vl%2F1ute7d2mAs%2BiVM43%2BQfhEITk%2Fg2oZv5r66Y%3D"
+$global:KubeletStartFile = $global:KubeDir + "\kubeletstart.ps1"
+$global:KubeProxyStartFile = $global:KubeDir + "\kubeproxystart.ps1"
+
+filter Timestamp {"$(Get-Date -Format o): $_"}
+
+function
+Write-Log($message)
+{
+    $msg = $message | Timestamp
+    Write-Output $msg
+}
+
+function
+Expand-ZIPFile($file, $destination)
+{
+    $shell = new-object -com shell.application
+    $zip = $shell.NameSpace($file)
+    foreach($item in $zip.items())
+    {
+        $shell.Namespace($destination).copyhere($item)
+    }
+}
+
+function
+Get-KubeBinaries()
+{
+    $zipfile = "c:\k.zip"
+    Invoke-WebRequest -Uri $global:KubeBinariesSASURL -OutFile $zipfile
+    Expand-ZIPFile -File $zipfile -Destination C:\
+}
+
+function
+Write-KubeConfig()
+{
+    $kubeConfigFile = $global:KubeDir + "\config"
+
+    $kubeConfig = @"
+---
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: "$CACertificate"
+    server: https://$MasterFQDNPrefix.$Location.cloudapp.azure.com
+  name: "$MasterFQDNPrefix"
+contexts:
+- context:
+    cluster: "$MasterFQDNPrefix"
+    user: "$MasterFQDNPrefix-admin"
+  name: "$MasterFQDNPrefix"
+current-context: "$MasterFQDNPrefix"
+kind: Config
+users:
+- name: "$MasterFQDNPrefix-admin"
+  user:
+    client-certificate-data: "$AgentCertificate"
+    client-key-data: "$AgentKey"
+"@
+
+    $kubeConfig | Out-File -encoding ASCII -filepath "$kubeConfigFile"    
+}
+
+function
+New-InfraContainer()
+{
+    cd $global:KubeDir
+    docker build -t kubletwin/pause . 
+}
+
+function
+Get-PodCIDR
+{
+    $argList = @("--hostname-override=$AzureHostname","--pod-infra-container-image=kubletwin/pause","--resolv-conf=""""","--api-servers=https://${MasterIP}:443","--kubeconfig=c:\k\config")
+    $process = Start-Process -FilePath c:\k\kubelet.exe -PassThru -ArgumentList $argList
+
+    $podCidrDiscovered=$false
+    $podCIDR=""
+    # run kubelet until podCidr is discovered
+    while (-not $podCidrDiscovered)

```

Comment:
Log a message here so the user knows that we're waiting on cidr. Otherwise looks like it will just silently loop/hang here.